### PR TITLE
Missing changes

### PR DESCRIPTION
**Pull request by WXbet**
- missing libdvbcsa changes
- build warning and error fixes

Don't forget to bump the emu version number, I guess to 800.

### CODE REVIEW

File: config.h
Comment (oscam-emu) [inline config.h:1]: Are this changes required?

Comment (WXbet) [reply config.h:1]: Both Options was activated before trunk sync, so I think yes.

Comment (ohmza) [reply config.h:1]: @oscam-emu These changes on config.h were introduced to emu on r799 but since you reverted all r799 changes, it has disappeared on latest merge.

I think oscam-emu is now required to have WITH_SSL enabled because of the inteoduction of BISS 2. Am I correct?

Comment (oscam-emu) [reply config.h:1]: If they were required for icam, which I reverted and now icam in on oscam trunk, then why these are not in trunk?

yes, biss 2 requires ssl

Comment (WXbet) [reply config.h:1]: icam on trunk does not need ssl to work.

Comment (ohmza) [reply config.h:1]: @oscam-emu trunk oscam doesn't require WITH_SSL to be enabled. However, if biss 2 required it why it has to be left disabled?

Comment (oscam-emu) [reply config.h:1]: nor radegast?

Comment (WXbet) [reply config.h:1]: This can be disabled, my fault.

**Comment by oscam-emu** [comment]
I think this is ok now. I only changed the coding style on some functions that use different declaration with and without emu.
I like it more that way, I think it's easier to understand how it is declared.

**Comment by oscam-emu** [comment]
@WXbet @ohmza if any of you want to get commit rights on this repo contact me via email.

**Comment by WXbet** [comment]
> I only changed the coding style on some functions that use different declaration with and without emu.
> I like it more that way, I think it's easier to understand how it is declared.

seems we have build errors:

```
CC module-emulator-powervu.c
module-emulator-powervu.c: In function 'powervu_ecm':
module-emulator-powervu.c:2166:128: error: invalid use of undefined type 'struct dvbcsa_bs_key_s'
2166 | dvbcsa_bs_key_set(cw[j], key_data[cdata->connid].key[j][EVEN]);
| ^
module-emulator-powervu.c:2166:128: error: invalid use of undefined type 'struct dvbcsa_bs_key_s'
2166 | dvbcsa_bs_key_set(cw[j], key_data[cdata->connid].key[j][EVEN]);
| ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
module-emulator-powervu.c:2170:128: error: invalid use of undefined type 'struct dvbcsa_bs_key_s'
2170 | dvbcsa_bs_key_set(cw[j], key_data[cdata->connid].key[j][ODD]);
| ^
module-emulator-powervu.c:2170:128: error: invalid use of undefined type 'struct dvbcsa_bs_key_s'
2170 | dvbcsa_bs_key_set(cw[j], key_data[cdata->connid].key[j][ODD]);
```

**Comment by oscam-emu** [comment]
Forced pushed, hopefully it's ok now.

**Comment by WXbet** [comment]
Yepp, build error is gone. But now we get warnings:
`module-streamrelay.c: In function 'write_cw':
module-streamrelay.c:194:63: warning: passing argument 3 of 'dvbcsa_bs_key_set_ecm' from incompatible pointer type [-Wincompatible-pointer-types]
  194 |                 dvbcsa_bs_key_set(er->cw, key_data[connid].key[0]);
      |                                           ~~~~~~~~~~~~~~~~~~~~^~~
      |                                                               |
      |                                                               struct dvbcsa_bs_key_s **
module-streamrelay.c:185:60: note: in definition of macro 'dvbcsa_bs_key_set'
  185 | #define dvbcsa_bs_key_set(a,b) dvbcsa_bs_key_set_ecm(ecm,a,b)
      |                                                            ^
In file included from module-streamrelay.h:31,
                 from module-streamrelay.c:7:
/home/wxbet/s3_releases/support/toolchains/dream_arm/arm-dreambox-linux-gnueabihf/sysroot/usr/include/dvbcsa/dvbcsa.h:99:99: note: expected 'struct dvbcsa_bs_key_s *' but argument is of type 'struct dvbcsa_bs_key_s **'
   99 | void dvbcsa_bs_key_set_ecm(const unsigned char ecm, const dvbcsa_cw_t cw, struct dvbcsa_bs_key_s *key);
      |                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~^~~
module-streamrelay.c:203:67: warning: passing argument 3 of 'dvbcsa_bs_key_set_ecm' from incompatible pointer type [-Wincompatible-pointer-types]
  203 |                 dvbcsa_bs_key_set(er->cw + 8, key_data[connid].key[0]);
      |                                               ~~~~~~~~~~~~~~~~~~~~^~~
      |                                                                   |
      |                                                                   struct dvbcsa_bs_key_s **
module-streamrelay.c:185:60: note: in definition of macro 'dvbcsa_bs_key_set'
  185 | #define dvbcsa_bs_key_set(a,b) dvbcsa_bs_key_set_ecm(ecm,a,b)
      |                                                            ^
/home/wxbet/s3_releases/support/toolchains/dream_arm/arm-dreambox-linux-gnueabihf/sysroot/usr/include/dvbcsa/dvbcsa.h:99:99: note: expected 'struct dvbcsa_bs_key_s *' but argument is of type 'struct dvbcsa_bs_key_s **'
   99 | void dvbcsa_bs_key_set_ecm(const unsigned char ecm, const dvbcsa_cw_t cw, struct dvbcsa_bs_key_s *key);
      |       `

**Comment by oscam-emu** [comment]
shit. I am exhausted right now. If you can fix it, by all means please do it. I sent you an invitation.

**Comment by WXbet** [comment]
Yes, give me a moment.

**Comment by tromoto** [comment]
great work guys! Thx!
